<?xml version="1.0"?>
<launch>
  <master auto="start"/>
  <arg name="namespace"/> 
  <arg name="init_pose_x"/> 
  <arg name="init_pose_y"/> 
  <arg name="init_pose_z"/> 
  <arg name="init_pose_yaw"/> 
  <param name="/use_sim_time" value="true"/> 

  <node pkg="tf" type="static_transform_publisher" name="static_tf_$(arg namespace)_to_odom" args="0.0 0.0 0.0 0.0 0.0 0.0 $(arg namespace)/map $(arg namespace)/odom  10"  /> 

  <node name="robot_laser_grid_mapping" pkg="multi_robot_exploration" type="grid_mapping.py" output="screen"> 
        <param name="sensor_model_p_occ"        value="0.75"/> 
        <param name="sensor_model_p_free"       value="0.45"/> 
        <param name="sensor_model_p_prior"      value="0.5"/> 
        <param name="robot_frame"               value="$(arg namespace)/base_link"/> 
        <param name="map_frame"                 value="$(arg namespace)/map"/> 
        <param name="map_center_x"              value="0.0"/> 
        <param name="map_center_y"              value="0.0"/> 
        <param name="map_size_x"                value="37.0"/> 
        <param name="map_size_y"                value="14.0"/> 
        <param name="map_resolution"            value="0.045"/> 
        <param name="map_publish_freq"          value="1.0"/> 
        <param name="update_movement"           value="0.1"/> <!-- update when moved this much -->
  </node>

  <node pkg="tf" type="static_transform_publisher" name="static_tf_$(arg namespace)_to_map" args="0.0 0.0 0.0 0.0 0.0 0.0 map $(arg namespace)/map 10"  />

  <group ns="/$(arg namespace)/map_merge">
      <param name="init_pose_x" value="$(arg init_pose_x)"/>
      <param name="init_pose_y" value="$(arg init_pose_y)"/>
      <param name="init_pose_z" value="$(arg init_pose_z)"/>
      <param name="init_pose_yaw" value="$(arg init_pose_yaw)"/>
  </group>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
    <remap from="$(arg namespace)/scan" to="scan" />
    <rosparam file="$(find multi_robot_exploration)/params/pioneer/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find multi_robot_exploration)/params/pioneer/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find multi_robot_exploration)/params/pioneer/costmap_common_params.yaml" command="load" ns="global_costmap"/>
    <rosparam file="$(find multi_robot_exploration)/params/pioneer/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find multi_robot_exploration)/params/pioneer/move_base_params.yaml" command="load" />
    <rosparam file="$(find multi_robot_exploration)/params/pioneer/local_planner_params.yaml" command="load" />
    <param name="TrajectoryPlannerROS/yaw_goal_tolerance" value="3.14" /> <!-- ignore yaw information in goal pose -->   
    <param name="global_costmap/robot_base_frame" value="$(arg namespace)/base_link"/>   
    <param name="local_costmap/global_frame" value="$(arg namespace)/map"/>
    <param name="TrajectoryPlannerROS/global_frame" value="$(arg namespace)/map"/>
    <param name="global_costmap/obstacles/scan/sensor_frame" value="$(arg namespace)/base_laser_link"/>
    <param name="global_costmap/obstacles/scan/topic" value="/$(arg namespace)/scan"/>
    <param name="local_costmap/robot_base_frame" value="$(arg namespace)/base_link"/> 
    <param name="local_costmap/obstacles/scan/sensor_frame" value="$(arg namespace)/base_laser_link"/>
    <param name="local_costmap/obstacles/scan/topic" value="/$(arg namespace)/scan"/>
  </node>

</launch>
